Oracle.md

```diff
@@ -0,0 +1,165 @@
+The oracle provides 6 R-values, meaning that it can attest to outcomes in the range `[0,999999]`:
+
+```
+base: 10
+RValues: [R0, R1, R2, R3, R4, R5]
+```
+
+Alice and Bob create a DLC with "relevant" outcomes between $10000 and $10999.
+Below $10000 Alice gets everything, and above $10999 Bob gets everything.
+They use the `R0` and `R1` values to cover the cases where BTC/USD < $10000 using `R0=0` and `R1=0`.
+They use `R0`, `R1` and `R2` to cover the cases where BTC/USD > $10999 and <= $19999, using `R1=0`, `R1=1` and `R2 in [0-9]`.
+They use `R0` and `R1` to cover the cases where BTC/USD >= $20000 and < $99999, using `R0=0` and `R1 in [2,9]`.
+Finally, they use only `R0` to cover the cases where BTC/USD >= $100000 and <= $999999, using `R0 in [1-9]`.
+
+At maturity time, if the BTC/USD rate is less than $999999, the oracle sign each of the digit representing the outcome value.
+If the rate is greater than $999999, the oracle signs the value $999999.
+
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`string`:`external_name`]
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`u16`:`num_outcomes`]
+   * [`u16`:`outcome_1_len`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`int32`:`start`]
+   * [`u32`:`count`]
+   * [`u16`:`step`]
+
+#### Version 0 `large_range_event_descriptor`
+
+1. type: 55306 (`large_range_event_descriptor_v0`)
+2. data:
+   * [`u16`:`base`]
+   * [`byte`:`is_signed`]
+   * [`u16`:`nb_nonces`]
+   * [`nb_nonces*x_point`:`oracle_nonces`]
+
+`is_signed` should be set to `0x00` for representing false and `0x01` for representing true.
+Any other value for it should be considered invalid.
+
+### Oracle events
+
+For users to be able to create DLCs based on a given event, they also need to obtain information about the oracle and the time at which it plans on releasing a signature over the event outcome.
+Oracle events contain such information, which includes:
+* the oracle public key,
+* the earliest time (UTC) at which it plans on releasing a signature over the event outcome, in epoch seconds,
+* the event descriptor,
+* the event URI which can be a name or categorization associated with the event by the oracle.
```

Comment:
I think we should require that the event URI specify the unit and precision of numbers for events which are signatures of numbers. Otherwise fraud proofs are hard.

E.g.
Unit: USD per Bitcoin
Precision: 10^-2 (which implies that the last digit signed is the 0.01 place, second to last digit signed is 0.1 and so on).

Reply:
I agree that this is needed, but I don't think that it makes sense to put that in an event URI. I think the two approaches would be to either:
* Add that to the range and large range event descriptors
* Add a metadata field where oracle can put this kind of things (and whatever else is relevant for a given event).

Reply:
yeah, I guess I don't really have strong feelings about where it should go so long as it is specified to be required

Reply:
added